docs/architecture/adr-008-blocksize-independent-commitment.md

```diff
@@ -0,0 +1,98 @@
+
+Following the non-interactive default rules, `minimalBlocksize` can be calculated like this:
+
+```go
+func minimalBlocksize(mLength uint64) uint64 {
+    blocksize := NextHigherPowerOf2(mLength)
+    //Check if message fits with non-interactive-deafult rules
+    if mLength <= (blocksize * (blocksize -1)) {
+        return blocksize
+    } else {
+        return blocksize << 1
+    }
+}
+```
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Negative
+
+1. The amount of subtroots per commitment increases from O(log(n)) to O(sqrt(n)) while n is the number of shares of a message. The worst case for the most amount of shares is depicted in the diagram below - an entire block missing one share.  
+  ![Interactive Commitment 2](./assets/complexity.png)
+
+2. With more subroots the amount of merkle proofs will increase. With deeper subroots the size of the merkle proofs will increase. So instead of having a merkle proof from the `DataRoot` to the `originalSubRoot` you will need the merkle proof from `DataRoot` to `2^k` amount of `miniSubRoots` with `k` being the height difference of `originalSubRoot` and `miniSubRoots`. You can optimize this by having one merkle proof from `DataRoot` to `originalSubRoot` and then `k` merkle proofs from `originalSubRoot` to the `miniSubRoots`. Because the merkle proof is created after the block is published, we know the block size and, therefore, if this `originalSubRoot` exists or not, like in smaller block sizes.
+
+### Positive
+
+1. A Rollup can include the commitment in the block header *before* posting on Celestia because it is size-independent and does not have to wait for Celestia to confirm the block size. In general, the rollup needs access to this commitment in some form to verify a message inclusion proof guaranteeing data availability, which Rollmint currently does not have access to.
+2. In turn, this would serve as an alternative to [ADR 007](https://github.com/celestiaorg/optimint/blob/main/docs/lazy-adr/adr-007-header-commit-to-shares.md)
+3. Here is one scheme on how a Rollup might use this new commitment in the block header. Let's assume a Rollup that looks like this:  
+  BH1 <-- BH2 <-- BH3 <-- BH4  
+  The Messages that are submitted to Celestia could look like this:  
+  Message 1: B1  
+  The Commitment of B1 is saved into BH1.  
+  Message 2: (BH1+B2)  
+  The Commitment of BH1+B2 is saved into BH2.  
+  Message 3: (BH2+B3)  
+  The Commitment of (BH2+B3) is saved into BH3, and so on.  
+4. Verifying a message inclusion proof could be done with merkle proofs of the subroots to the `DataRoot` and then recalculating the commitment and comparing to what's in the rollup block header. If you trust the validator set of Celestia/Rollup (aka Rollmint Light Client) it could be as simple as submitting a proof over the pfd transaction that included the message and then check if the commitment is the same as in the pfd transaction.
```

Comment:
```suggestion
4. Verifying a message inclusion proof could be done with merkle proofs of the subroots to the `DataRoot` and then recalculating the commitment and comparing to what's in the rollup block header. If you trust the validator set of Celestia/Rollup (aka Rollmint Light Client) it could be as simple as submitting a proof over the PFD transaction that included the message and then check if the commitment is the same as in the PFD transaction.
```